app/scripts/models/auth_brokers/redirect.js

```diff
@@ -19,15 +20,23 @@ define([
       return p()
         .then(function () {
           var redirectTo = result.redirect;
+          var params = {};
+
+          var startOfParams = redirectTo.lastIndexOf('?');
```

Comment:
Maybe extract the URL->search params section into lib/url.js? It certainly could be reused in the future, and it'll hopefully reduce the amount of code in this function.